From da9ee449b32b89f54e7389c50f21a46655572b7c Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 20 Sep 2021 15:06:18 +0300
Subject: [PATCH 3/3] clk/s32g: Remove unnecessary partition blocks for cores

MC_ME A53 cores have artificially defined blocks with no effect
in the code. All of these are removed by the current commit.

Upstream-Status: Pending

Issue: ALB-7724
Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32g/clk/enable_clk.c                  |  3 ---
 drivers/nxp/s32g/clk/s32gen1_clk.c                 | 10 +---------
 include/drivers/nxp/s32g/clk/s32gen1_clk_modules.h |  4 ----
 3 files changed, 1 insertion(+), 16 deletions(-)

diff --git a/drivers/nxp/s32g/clk/enable_clk.c b/drivers/nxp/s32g/clk/enable_clk.c
index 3b168e01c..31d06c103 100644
--- a/drivers/nxp/s32g/clk/enable_clk.c
+++ b/drivers/nxp/s32g/clk/enable_clk.c
@@ -287,9 +287,6 @@ static int enable_part_block(struct s32gen1_clk_obj *module,
 	uint32_t cofb;
 
 	switch (block->block) {
-	case s32gen1_part_core0 ... s32gen1_part_core3:
-		/* Nothing to do here */
-		break;
 	case s32gen1_part_block0 ... s32gen1_part_block15:
 		cofb = block->block - s32gen1_part_block0;
 		if (enable)
diff --git a/drivers/nxp/s32g/clk/s32gen1_clk.c b/drivers/nxp/s32g/clk/s32gen1_clk.c
index 50d8cbdc1..250f02403 100644
--- a/drivers/nxp/s32g/clk/s32gen1_clk.c
+++ b/drivers/nxp/s32g/clk/s32gen1_clk.c
@@ -98,16 +98,8 @@ static struct s32gen1_clk cgm1_mux0_clk =
 		S32GEN1_MODULE_CLK(cgm1_mux0);
 
 /* A53_CORE */
-static struct s32gen1_part_block arm_core0_block =
-		S32GEN1_PART_BLOCK(cgm1_mux0_clk, 1, s32gen1_part_core0);
-static struct s32gen1_part_block arm_core1_block =
-		S32GEN1_PART_BLOCK(arm_core0_block, 1, s32gen1_part_core1);
-static struct s32gen1_part_block arm_core2_block =
-		S32GEN1_PART_BLOCK(arm_core1_block, 1, s32gen1_part_core2);
-static struct s32gen1_part_block arm_core3_block =
-		S32GEN1_PART_BLOCK(arm_core2_block, 1, s32gen1_part_core3);
 static struct s32gen1_clk a53_core_clk =
-		S32GEN1_FREQ_MODULE_CLK(arm_core3_block, 0,
+		S32GEN1_FREQ_MODULE_CLK(cgm1_mux0_clk, 0,
 					S32GEN1_A53_MAX_FREQ);
 
 /* A53_CORE_DIV2 */
diff --git a/include/drivers/nxp/s32g/clk/s32gen1_clk_modules.h b/include/drivers/nxp/s32g/clk/s32gen1_clk_modules.h
index a0c42a8c1..4993f2f4d 100644
--- a/include/drivers/nxp/s32g/clk/s32gen1_clk_modules.h
+++ b/include/drivers/nxp/s32g/clk/s32gen1_clk_modules.h
@@ -172,10 +172,6 @@ enum s32gen1_clkm_type {
 };
 
 enum s32gen1_part_block_type {
-	s32gen1_part_core0,
-	s32gen1_part_core1,
-	s32gen1_part_core2,
-	s32gen1_part_core3,
 	s32gen1_part_block0,
 	s32gen1_part_block1,
 	s32gen1_part_block2,
-- 
2.17.1

